mozilla-central/setup

```diff
@@ -153,48 +183,69 @@ if [ -n "$INDEXED_GIT_REV" ]; then
-        # analysis data might refer to line numbers that don't exist. For now we only copy
-        # files from the generated-$PLATFORM folder into objdir/ if the file doesn't
-        # already exist there. Effectively this means that if the same generated file
-        # exists for multiple platforms, but has different contents, we will prefer the
-        # one from the platform encountered earliest in this loop.
-        # See bug 1487583 comment 3 for some more discussion.
         pushd generated-$PLATFORM
-        set +x  # Turn off echoing of commands and output only relevant things to avoid bloating logfiles
-        find . -type f |
-        while read GENERATED_FILE; do
-            if [ ! -f "../objdir/$GENERATED_FILE" ]; then
-                # Generated file wasn't encountered for previously processed platforms
-                echo "Taking generated file $GENERATED_FILE from $PLATFORM"
-                move_file "$GENERATED_FILE" "../objdir/$GENERATED_FILE"
-                if [ -f "../analysis-$PLATFORM/__GENERATED__/$GENERATED_FILE" ]; then
-                    # Move the analysis file as well
-                    move_file "../analysis-$PLATFORM/__GENERATED__/$GENERATED_FILE" "../analysis/__GENERATED__/$GENERATED_FILE"
-                fi
-            fi
-        done
-        set -x
+        find . -type f >> ../generated-files.list
         popd
 
-        # Throw away any leftover generated files and their analysis, so that they don't
-        # accidentally get merged later. The above loop should have copied any that we
-        # care about into the main objdir/ and analysis/ folders.
-        rm -rf generated-$PLATFORM
-        rm -rf analysis-$PLATFORM/__GENERATED__
-
         # List all the analysis files we have left. We will merge these across platforms
-        # after this per-platform loop is complete.
+        # after this per-platform loop is complete. Make sure to skip over the __GENERATED__
+        # directory
         pushd analysis-$PLATFORM
-        find . -type d >> ../analysis-dirs.list
-        find . -type f >> ../analysis-files.list
+        find . -not \( -name __GENERATED__ -prune \) -type d >> ../analysis-dirs.list
+        find . -not \( -name __GENERATED__ -prune \) -type f >> ../analysis-files.list
         popd
 
     done    # end PLATFORM loop
 
     date
 
+    # Special case: xptdata.cpp is a giant file and is different for each platform, but
+    # the differences are not particularly relevant so let's just keep the linux one.
+    rm -f              generated-macosx64/xpcom/reflect/xptinfo/xptdata.cpp
+    rm -f                 generated-win64/xpcom/reflect/xptinfo/xptdata.cpp
+    rm -f analysis-macosx64/__GENERATED__/xpcom/reflect/xptinfo/xptdata.cpp
+    rm -f    analysis-win64/__GENERATED__/xpcom/reflect/xptinfo/xptdata.cpp
+
+    # For each generated file, if all platforms generated the same thing (or didn't
+    # generate the file at all), copy it to the merged objdir.
```

Comment:
What sort of files are only generated on some platforms? It feels like it could be confusing but maybe it is obvious enough from the file.

Reply:
There's some stuff like `gfx/layers/CompositorD3D11Shaders.h` and `accessible/interfaces/ia2/Accessible2.h` which is windows-only. And stuff like `build/unix/elfhack/inject/x86_64.c` is Linux-only. It is mostly obvious from the file in question.